server/controllers/recipe.js

```diff
@@ -0,0 +1,323 @@
+// import module dependencies
+import models from '../models';
+import pagenate from '../helper/pagenate';
+
+// create reference db model
+const Recipe = models.Recipe;
+
+/**
+ * query: hold query limit and offset
+ */
+const query = {};
+
+export default {
+/**
+   * @description add recipe controller
+   * @param {Object} req - Request object
+   * @param {Object} res - Response object
+   * @returns {Object} json - payload
+   */
+  addRecipe(req, res) {
+    Recipe
+      .create({
+        title: req.body.title,
+        description: req.body.description,
+        ingredients: req.body.ingredients,
+        procedures: req.body.procedures,
+        imageUrl: req.body.imageUrl,
+        addedBy: req.decoded.user.id,
+      })
+      .then(recipe => res.status(201).send({
+        message: 'Added successfully',
+        recipe,
+      }))
+      .catch(error => res.status(400).send({ message: error.message }));
+  },
+
+  /**
+     * @description get all recipe controller
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @returns {Object} json - payload
+     */
+  getAllRecipe(req, res) {
+    /**
+     * query limit: get query limit if supplie else use default
+     * query offset: get query offset if supplie else use default
+     */
+    query.limit = req.query.limit || 1;
+    query.offset = req.query.offset || 0;
+
+    /**
+     * query the db for all recipes
+     * left join Reviews as reviews
+     * left join Favourites as favourites
+     * left join left join Votings as votings
+     * ordered by 'id' descending
+     */
+    Recipe
+      .findAndCountAll({
+        order: [['id', 'DESC']],
+        limit: query.limit,
+        offset: query.offset,
+      })
+      .then((recipe) => {
+        if (recipe.rows.length <= 0) {
+          return res.status(404).send({ message: 'No recipe(s) was found!' });
```

Comment:
Nitpick.
Since the check is that the not even one recipe was found: the appropriate message should be `No recipe was found!`